Ahmedq02/task-app
# Improve auth

# Improve authentication

## Description
Migrated from the pre-existing hard-coded password authentication to a custom Django authentication that has the same behaviour. This will enable the protection of future endpoints

- **Related Issue:** Fixes #5 
- **Summary of Changes:**
  - Removed hard-coded password check from views
  - Created new endpoints for login and logout using only a password
  - Created a custom Django auth system (backend and user manager) for managing passkeys
  - Protected the endpoints with the new authentication system

## How to Test



1. Check out this branch: `git checkout fix-auth`
2. Run `poetry install`
3. Drop database (needed for creating the custom users passkey)
4. Apply migrations: `make migrate`
5. Add .env file with `STANDARD_USER_USERNAME` variable
6. Run the development server: `make runserver`
7. Navigate to the admin page, and create a User - the username will be auto-populated with the `STANDARD_USER_USERNAME` (this must be kept for the passkey)
8. Test the following:
   - [x] Navigating to Dashboard is passkey protected
   - [x] Navigating to Add Task is passkey protected
   - [x] Logging in keeps you logged in

## Deployment Notes



- [x] This PR requires database migrations
- [x] New environment variables need to be added: `STANDARD_USER_USERNAME` (describes the name of the standard user used for storing the password in the database)
